### ISSUE TITLE
Questions about eval_generic.py

### ISSUE THREAD

**Issue by kevinhartyanyi** [hkchengrex/STCN#111]
Hi, thanks for sharing this amazing work.

I was playing around a little with the [eval_generic.py](https://github.com/hkchengrex/STCN/blob/main/eval_generic.py) code, and I have a few questions about it.

1. Multiple ground truth images: Can I provide multiple annotated images for the same object on different frames? Inside the code, the comment says that we need to provide the first appearance of each object. I also see [here](https://github.com/hkchengrex/STCN/blob/79cda2deec17b1f9993e28f560b543adb7200c15/dataset/generic_test_dataset.py#L78) that the generic_dataset can take multiple ground truth masks and if I understood the code correctly then it decides if it's a new object based on the color of the mask. Does this mean that using the same color for the same object on different frames I can provide those frames to the model to increase the accuracy?
2. Backward propagation: Can I run the model in backward propagation as well? So if I don't have annotations for an object on frame 1 and 2, but I have it on frame 3, then it would try to make an estimation on frame 1 and 2 as well not only on the frames after frame 3.

Basically what I want to do is this: Let's say I have a video that is 10 frames long, and I have the same object annotated on frame 3, 5 and 8. I would like to use the model to make an estimation based on these already annotated frames for the rest of the frames. 

Thanks in advance.

**Comment by hkchengrex** [comment]
The current code supports data in the YouTubeVOS style, so new ground-truth frames can only contain new objects.

There are two ways to do this:
1. [Interactive VOS](https://github.com/hkchengrex/MiVOS/tree/MiVOS-STCN), where the frames are inputted sequentially. It will perform propagation independently for each frame and try to resolve the conflicts between different propagations.
2. Change `eval_generic`, the dataset, and `inference_core_yv` such that multiple ground truth masks are loaded at the very beginning and propagate just once.
